firmware/Components.h

```diff
@@ -68,92 +71,94 @@ namespace platypus
     void arm();
   };
 
-  // Sensors //
-  class AnalogSensor : public Sensor 
+  // Internal Sensors //
+  class BatterySensor : public Sensor
   {
   public:
-    AnalogSensor(int channel);
+    BatterySensor(int id);
+    virtual char* name();
+    void loop();
 
-    bool set(const char* param, const char* value);
-    virtual char *name() = 0;
-    
-    void scale(float scale);
-    float scale();
-    
-    void offset(float offset);
-    float offset();
-    
   private:
-    float scale_;
-    float offset_;
+    const int measurementInterval;
```

Comment:
Might just rename this to `interval`, because it's the only timing related thing in here, but that's just a preference.  This is a fine name too.